CMakeLists.txt

```diff
@@ -27,6 +27,7 @@ option(ENABLE_CCACHE "Speed up incremental rebuilds via ccache" ON)
 option(ENABLE_COVERAGE "Build with coverage instrumentalisation" OFF)
 option(ENABLE_COMPILER_WARNINGS "Build with compiler warnings" OFF)
 option(ENABLE_SANITIZER "Use memory sanitizer for Debug build" OFF)
```

Comment:
I would rather rename this option if we choose to have separate flags for each sanitizer. Sth like 
```suggestion
option(ENABLE_ADDRESS_SANITIZER "Use memory sanitizer for Debug build" OFF)
```

Reply:
Thats a good suggestion. We can also have a top-level sanitizer flag that automatically enables all the sub-flags. So users can either switch on all flags sanitizer flags at once by enabling the top-level flag, or select specific ones they want.

Reply:
We also need to update the CMake build flags table in the top-level README.

Reply:
Hmmm, one flag to rule them all... Done 🙂